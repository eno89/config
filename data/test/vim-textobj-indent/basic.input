" Tests for textobj-indent
function s:test_global_variables()  "{{{1
  Title "Check global variables"

  Assert !!exists('g:loaded_textobj_indent') ==> 1
endfunction




function s:test_named_key_mappings()  "{{{1
  Title "Check the existence of the named key mappings in each mode"

  for _ in ['<Plug>(textobj-indent-a)',
  \         '<Plug>(textobj-indent-i)',
  \         '<Plug>(textobj-indent-same-a)',
  \         '<Plug>(textobj-indent-same-i)']
    execute "Assert maparg(".string(_).", 'c') != '' ==> 0"
    execute "Assert maparg(".string(_).", 'i') != '' ==> 0"
    execute "Assert maparg(".string(_).", 'n') != '' ==> 0"
    execute "Assert maparg(".string(_).", 'o') != '' ==> 1"
    execute "Assert maparg(".string(_).", 'v') != '' ==> 1"
  endfor
endfunction




function s:test_default_key_mappings()  "{{{1
  Title "Check the definitions of the default key mappings in each mode"

  Assert maparg('ai', 'c') ==> ''
  Assert maparg('ai', 'i') ==> ''
  Assert maparg('ai', 'n') ==> ''
  Assert maparg('ai', 'o') ==> '<Plug>(textobj-indent-a)'
  Assert maparg('ai', 'v') ==> '<Plug>(textobj-indent-a)'
  Assert maparg('ii', 'c') ==> ''
  Assert maparg('ii', 'i') ==> ''
  Assert maparg('ii', 'n') ==> ''
  Assert maparg('ii', 'o') ==> '<Plug>(textobj-indent-i)'
  Assert maparg('ii', 'v') ==> '<Plug>(textobj-indent-i)'
  Assert maparg('aI', 'c') ==> ''
  Assert maparg('aI', 'i') ==> ''
  Assert maparg('aI', 'n') ==> ''
  Assert maparg('aI', 'o') ==> '<Plug>(textobj-indent-same-a)'
  Assert maparg('aI', 'v') ==> '<Plug>(textobj-indent-same-a)'
  Assert maparg('iI', 'c') ==> ''
  Assert maparg('iI', 'i') ==> ''
  Assert maparg('iI', 'n') ==> ''
  Assert maparg('iI', 'o') ==> '<Plug>(textobj-indent-same-i)'
  Assert maparg('iI', 'v') ==> '<Plug>(textobj-indent-same-i)'
endfunction




function s:test_select_a()  "{{{1
  Title "<Plug>(textobj-indent-a)"

  tabnew
  tabonly!

  silent put =[
  \   'if some_condition_is_satisfied',
  \   '  if another_condition_is_satisfied',
  \   '    call s:special_stuff()',
  \   '  endif',
  \   '  call s:normal_stuff()',
  \   '',
  \   '  ...',
  \   '  endif',
  \   'else',
  \   '  ...',
  \   'endif',
  \ ]
  1 delete _
  normal! 4G

  execute "normal v\<Plug>(textobj-indent-a)\<Esc>"
  Assert [line("'<"), col("'<")] ==> [2, 1]
  Assert [line("'>"), col("'>")] ==> [8, 8]

  execute "silent normal y\<Plug>(textobj-indent-a)"
  Assert [line("'["), col("'[")] ==> [2, 1]
  Assert [line("']"), col("']")] ==> [8, 8]
endfunction




function s:test_select_i()  "{{{1
  Title "<Plug>(textobj-indent-i)"

  tabnew
  tabonly!

  silent put =[
  \   'if some_condition_is_satisfied',
  \   '  if another_condition_is_satisfied',
  \   '    call s:special_stuff()',
  \   '  endif',
  \   '  call s:normal_stuff()',
  \   '',
  \   '  ...',
  \   '  endif',
  \   'else',
  \   '  ...',
  \   'endif',
  \ ]
  1 delete _
  normal! 4G

  execute "normal v\<Plug>(textobj-indent-i)\<Esc>"
  Assert [line("'<"), col("'<")] ==> [2, 1]
  Assert [line("'>"), col("'>")] ==> [5, 24]

  execute "silent normal y\<Plug>(textobj-indent-i)"
  Assert [line("'["), col("'[")] ==> [2, 1]
  Assert [line("']"), col("']")] ==> [5, 24]
endfunction




function s:test_select_same_a()  "{{{1
  Title "<Plug>(textobj-indent-same-a)"

  tabnew
  tabonly!

  silent put =[
  \   'if some_condition_is_satisfied',
  \   '  if another_condition_is_satisfied',
  \   '    call s:special_stuff()',
  \   '  endif',
  \   '  call s:normal_stuff()',
  \   '',
  \   '  ...',
  \   '  endif',
  \   'else',
  \   '  ...',
  \   'endif',
  \ ]
  1 delete _
  normal! 4G

  execute "normal v\<Plug>(textobj-indent-same-a)\<Esc>"
  Assert [line("'<"), col("'<")] ==> [4, 1]
  Assert [line("'>"), col("'>")] ==> [8, 8]

  execute "silent normal y\<Plug>(textobj-indent-same-a)"
  Assert [line("'["), col("'[")] ==> [4, 1]
  Assert [line("']"), col("']")] ==> [8, 8]
endfunction




function s:test_select_same_i()  "{{{1
  Title "<Plug>(textobj-indent-same-i)"

  tabnew
  tabonly!

  silent put =[
  \   'if some_condition_is_satisfied',
  \   '  if another_condition_is_satisfied',
  \   '    call s:special_stuff()',
  \   '  endif',
  \   '  call s:normal_stuff()',
  \   '',
  \   '  ...',
  \   '  endif',
  \   'else',
  \   '  ...',
  \   'endif',
  \ ]
  1 delete _
  normal! 4G

  execute "normal v\<Plug>(textobj-indent-same-i)\<Esc>"
  Assert [line("'<"), col("'<")] ==> [4, 1]
  Assert [line("'>"), col("'>")] ==> [5, 24]

  execute "silent normal y\<Plug>(textobj-indent-same-i)"
  Assert [line("'["), col("'[")] ==> [4, 1]
  Assert [line("']"), col("']")] ==> [5, 24]
endfunction




Test  "{{{1

" __END__
" vim: filetype=vim foldmethod=marker
