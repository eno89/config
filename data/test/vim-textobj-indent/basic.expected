===== "Check the definitions of the default key mappings in each mode"
TEST: maparg('ai', 'c')  ==>  '' ... ok
TEST: maparg('ai', 'i')  ==>  '' ... ok
TEST: maparg('ai', 'n')  ==>  '' ... ok
TEST: maparg('ai', 'o')  ==>  '<Plug>(textobj-indent-a)' ... ok
TEST: maparg('ai', 'v')  ==>  '<Plug>(textobj-indent-a)' ... ok
TEST: maparg('ii', 'c')  ==>  '' ... ok
TEST: maparg('ii', 'i')  ==>  '' ... ok
TEST: maparg('ii', 'n')  ==>  '' ... ok
TEST: maparg('ii', 'o')  ==>  '<Plug>(textobj-indent-i)' ... ok
TEST: maparg('ii', 'v')  ==>  '<Plug>(textobj-indent-i)' ... ok
TEST: maparg('aI', 'c')  ==>  '' ... ok
TEST: maparg('aI', 'i')  ==>  '' ... ok
TEST: maparg('aI', 'n')  ==>  '' ... ok
TEST: maparg('aI', 'o')  ==>  '<Plug>(textobj-indent-same-a)' ... ok
TEST: maparg('aI', 'v')  ==>  '<Plug>(textobj-indent-same-a)' ... ok
TEST: maparg('iI', 'c')  ==>  '' ... ok
TEST: maparg('iI', 'i')  ==>  '' ... ok
TEST: maparg('iI', 'n')  ==>  '' ... ok
TEST: maparg('iI', 'o')  ==>  '<Plug>(textobj-indent-same-i)' ... ok
TEST: maparg('iI', 'v')  ==>  '<Plug>(textobj-indent-same-i)' ... ok
Result: 20 / 20

===== "Check global variables"
TEST: !!exists('g:loaded_textobj_indent')  ==>  1 ... ok
Result: 1 / 1

===== "Check the existence of the named key mappings in each mode"
TEST: maparg('<Plug>(textobj-indent-a)', 'c') != ''  ==>  0 ... ok
TEST: maparg('<Plug>(textobj-indent-a)', 'i') != ''  ==>  0 ... ok
TEST: maparg('<Plug>(textobj-indent-a)', 'n') != ''  ==>  0 ... ok
TEST: maparg('<Plug>(textobj-indent-a)', 'o') != ''  ==>  1 ... ok
TEST: maparg('<Plug>(textobj-indent-a)', 'v') != ''  ==>  1 ... ok
TEST: maparg('<Plug>(textobj-indent-i)', 'c') != ''  ==>  0 ... ok
TEST: maparg('<Plug>(textobj-indent-i)', 'i') != ''  ==>  0 ... ok
TEST: maparg('<Plug>(textobj-indent-i)', 'n') != ''  ==>  0 ... ok
TEST: maparg('<Plug>(textobj-indent-i)', 'o') != ''  ==>  1 ... ok
TEST: maparg('<Plug>(textobj-indent-i)', 'v') != ''  ==>  1 ... ok
TEST: maparg('<Plug>(textobj-indent-same-a)', 'c') != ''  ==>  0 ... ok
TEST: maparg('<Plug>(textobj-indent-same-a)', 'i') != ''  ==>  0 ... ok
TEST: maparg('<Plug>(textobj-indent-same-a)', 'n') != ''  ==>  0 ... ok
TEST: maparg('<Plug>(textobj-indent-same-a)', 'o') != ''  ==>  1 ... ok
TEST: maparg('<Plug>(textobj-indent-same-a)', 'v') != ''  ==>  1 ... ok
TEST: maparg('<Plug>(textobj-indent-same-i)', 'c') != ''  ==>  0 ... ok
TEST: maparg('<Plug>(textobj-indent-same-i)', 'i') != ''  ==>  0 ... ok
TEST: maparg('<Plug>(textobj-indent-same-i)', 'n') != ''  ==>  0 ... ok
TEST: maparg('<Plug>(textobj-indent-same-i)', 'o') != ''  ==>  1 ... ok
TEST: maparg('<Plug>(textobj-indent-same-i)', 'v') != ''  ==>  1 ... ok
Result: 20 / 20

===== "<Plug>(textobj-indent-a)"
TEST: [line("'<"), col("'<")]  ==>  [2, 1] ... ok
TEST: [line("'>"), col("'>")]  ==>  [8, 8] ... ok
TEST: [line("'["), col("'[")]  ==>  [2, 1] ... ok
TEST: [line("']"), col("']")]  ==>  [8, 8] ... ok
Result: 4 / 4

===== "<Plug>(textobj-indent-i)"
TEST: [line("'<"), col("'<")]  ==>  [2, 1] ... ok
TEST: [line("'>"), col("'>")]  ==>  [5, 24] ... ok
TEST: [line("'["), col("'[")]  ==>  [2, 1] ... ok
TEST: [line("']"), col("']")]  ==>  [5, 24] ... ok
Result: 4 / 4

===== "<Plug>(textobj-indent-same-a)"
TEST: [line("'<"), col("'<")]  ==>  [4, 1] ... ok
TEST: [line("'>"), col("'>")]  ==>  [8, 8] ... ok
TEST: [line("'["), col("'[")]  ==>  [4, 1] ... ok
TEST: [line("']"), col("']")]  ==>  [8, 8] ... ok
Result: 4 / 4

===== "<Plug>(textobj-indent-same-i)"
TEST: [line("'<"), col("'<")]  ==>  [4, 1] ... ok
TEST: [line("'>"), col("'>")]  ==>  [5, 24] ... ok
TEST: [line("'["), col("'[")]  ==>  [4, 1] ... ok
TEST: [line("']"), col("']")]  ==>  [5, 24] ... ok
Result: 4 / 4

===== Total
Result: 57 / 57
